crates/biome_js_formatter/src/js/expressions/call_arguments.rs

```diff
@@ -508,9 +508,14 @@ fn write_grouped_arguments(
         buffer.into_vec().into_boxed_slice()
     };
 
-    if grouped_breaks {
+    // If the grouped content breaks, then we can skip the most_flat variant,
+    // since we already know that it won't be fitting on a single line.
+    let variants = if grouped_breaks {
         write!(f, [expand_parent()])?;
-    }
+        vec![middle_variant, most_expanded.into_boxed_slice()]
+    } else {
+        vec![most_flat, middle_variant, most_expanded.into_boxed_slice()]
```

Comment:
not sure why, but the `best_fitting!` macro really doesn't like playing along here with `Box<[FormatElement]>`, otherwise i'd prefer to use that than the unsafe block below.